Sources/Basics/Triple.swift

```diff
@@ -10,173 +10,36 @@
-
-        public init?(rawValue: String) {
-            if rawValue.hasPrefix(ABI.android.rawValue) {
-                self = .android
-            } else if let version = rawValue.firstIndex(where: { $0.isNumber }) {
-                self = .other(name: String(rawValue[..<version]))
-            } else {
-                self = .other(name: rawValue)
-            }
-        }
-
-        public var rawValue: String {
-            switch self {
-            case .android: return "android"
-            case .other(let name): return name
-            case .unknown: return "unknown"
-            }
-        }
-
-        public static func == (lhs: ABI, rhs: ABI) -> Bool {
-            switch (lhs, rhs) {
-            case (.unknown, .unknown):
-                return true
-            case (.android, .android):
-                return true
-            case (.other(let lhsName), .other(let rhsName)):
-                return lhsName == rhsName
-            default:
-                return false
-            }
-        }
-    }
-
-    public init(_ string: String) throws {
-        let components = string.split(separator: "-").map(String.init)
-
-        guard components.count == 3 || components.count == 4 else {
-            throw Error.badFormat(triple: string)
-        }
-
-        guard let arch = Arch(rawValue: components[0]) else {
-            throw Error.unknownArch(arch: components[0])
-        }
-
-        let vendor = Vendor(rawValue: components[1]) ?? .unknown
-
-        guard let os = Triple.parseOS(components[2]) else {
-            throw Error.unknownOS(os: components[2])
-        }
-
-        let osVersion = Triple.parseVersion(components[2])
-
-        let abi = components.count > 3 ? Triple.ABI(rawValue: components[3]) : nil
-        let abiVersion = components.count > 3 ? Triple.parseVersion(components[3]) : nil
+public typealias Triple = SwiftDriver.Triple
 
-        self.tripleString = string
-        self.arch = arch
-        self.vendor = vendor
-        self.os = os
-        self.osVersion = osVersion
-        self.abi = abi ?? .unknown
-        self.abiVersion = abiVersion
-    }
-
-    fileprivate static func parseOS(_ string: String) -> OS? {
-        var candidates = OS.allCases.map { (name: $0.rawValue, value: $0) }
-        // LLVM target triples support this alternate spelling as well.
-        candidates.append((name: "macos", value: .macOS))
-        return candidates.first(where: { string.hasPrefix($0.name) })?.value
+extension Triple {
+    public init(_ description: String) throws {
+        self.init(description, normalizing: false)
     }
+}
 
-    fileprivate static func parseVersion(_ string: String) -> String? {
-        let candidate = String(string.drop(while: { $0.isLetter }))
-        if candidate != string && !candidate.isEmpty {
-            return candidate
-        }
-
-        return nil
-    }
+extension Triple {
+    public static let macOS = try! Self("x86_64-apple-macosx")
```

Comment:
does Triple need a non-throwing constructer that takes a static string (can precondition)

Reply:
Probably, but I want to avoid additional changes API changes in this PR. (The existing triple uses the above API shape)